### PR TITLE
add DashScope qwen llms and multi modal llms qwen-vl

### PR DESCRIPTION
**Pull request by liuyhwangyh**
# Description

Support [DashScope](https://dashscope.aliyun.com/) qwen llms and multi modal llm qwen-vl.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/examples/llm/dashscope.ipynb [hunk 1]
```diff
@@ -0,0 +1,1768 @@
+{
```
Comment (hatianzhang) [inline docs/examples/llm/dashscope.ipynb:1]: <p>we should output delta of response.</p><p>rn it outputs the previous msg + delta which is not correct</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10385/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: docs/examples/multi_modal/dashscope_multi_modal.ipynb [hunk 2]
```diff
@@ -0,0 +1,325 @@
+{
```
Comment (hatianzhang) [inline docs/examples/multi_modal/dashscope_multi_modal.ipynb:1]: <p>nit can remove </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10385/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (hatianzhang) [inline docs/examples/multi_modal/dashscope_multi_modal.ipynb:1]: <p>nit can remove </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/10385/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: llama_index/multi_modal_llms/dashscope_utils.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+from llama_index.core.llms.types import (
+    ChatMessage,
+    ChatResponse,
+    CompletionResponse,
+)
+from llama_index.schema import ImageDocument
+
+
+def dashscope_response_to_completion_response(
+    response: Any, stream: bool = False
```
Comment (hatianzhang) [inline llama_index/multi_modal_llms/dashscope_utils.py:14]: is `stream` been used here

Comment (liuyhwangyh) [reply llama_index/multi_modal_llms/dashscope_utils.py]: removed

File: docs/examples/llm/dashscope.ipynb (hunk 1 above)
Comment (liuyhwangyh) [reply docs/examples/llm/dashscope.ipynb:1]: fix

File: docs/examples/multi_modal/dashscope_multi_modal.ipynb (hunk 2 above)
Comment (liuyhwangyh) [reply docs/examples/multi_modal/dashscope_multi_modal.ipynb:1]: updated

**Review by hatianzhang — APPROVED** [review]

**Pull request merged**